docs/src/tutorial/embedding.rst

```diff
@@ -58,30 +57,123 @@ to export it as a linker symbol that can be used by other C files, which in this
-If you want to generate a portable application we recommend using a specialized
-tool (e.g. `PyInstaller <https://pyinstaller.org/en/stable/>`_
-or `cx_freeze <https://cx-freeze.readthedocs.io/en/latest/index.html>`_) to find and
-bundle these dependencies.
+Cython generate one into your module's C file with the ``cython --embed`` option.
+
+When using ``--embed``, Cython generates the C implementation file (e.g., ``embedded.c``) along with a header file (e.g., ``embedded.h``).
+
+The generated ``embedded.c`` file contains a ``__Pyx_main()`` function. This function initializes the Python interpreter and can be modified to call public functions exported by your Cython module (such as ``say_hello_from_python()``).
+
+The full process for compiling and running is as follows:
+
+.. code-block:: bash
+
+    gcc -c embedded.c $(python3-config --cflags) -o embedded.o
+    gcc embedded.o $(python3-config --ldflags --embed) -o embedded
+
+After compilation, the ``embedded`` executable is created. Running it will execute the content of ``__Pyx_main()``:
+
+.. code-block:: bash
+
+    $ ./embedded
+    Hello from Python!
+
+
+Embedding multiple modules
+==========================
+
+To statically link multiple Cython modules into a single executable, you must use the
+``cython --embed-modules`` option alongside the ``cython --embed`` option.
+This option enables the necessary C-level initialisation calls (via :c:func:`PyImport_AppendInittab`)
+for all listed modules within the main program's C code.
+
+The argument is a comma-separated list of the module names you wish to embed:
+
+.. code-block:: bash
+
+    cython --embed --embed-modules=lcmath,another_module main_script.pyx
+
+To illustrate how it works, let's assume we have the following files:
+
+.. code-block:: cython
+   :caption: lcmath.pyx
+
+    def add(a, b):
+        return a + b
+
+    cdef int sub(int a, int b):
+        return a - b
+
+.. code-block:: cython
+   :caption: lcmath.pxd
+
+    cdef int sub(int a, int b)
+
+.. code-block:: cython
+   :caption: combinatorics.pyx
+
+    from lcmath import add
+    from lcmath cimport sub
+
+    cdef public void add_one(int a):
+        print(add(a, 1))
+
+    cdef public void sub_one(int a):
+        print(sub(a, 1))
+
+The full process for static linking involves these steps:
+
+First, generate the dependency C code:
+
+.. code-block:: bash
+
+        cython lcmath.pyx  # creates lcmath.c
+
+Next, generate the main C code(which includes the ``main()`` function and module initialisers):
+
+.. code-block:: bash
+
+        cython --embed --embed-modules=lcmath combinatorics.pyx  # creates combinatorics.c
+
+.. note::
+    Since ``combinatorics.pyx`` is the main module, the generated ``combinatorics.c`` will contain the main function. You must manually update this C file (e.g., adding calls to ``add_one()`` and ``sub_one()``) within its ``__Pyx_main()`` function to execute your code.
```

Comment:
Erm, no? You can simply add executable script code to your module, as you would in Python. It will then get executed when importing the module that executes the C `main()` function.

The `sub_one` and `add_one` functions also don't need to be `public` for that. I think it's confusing you present it that way. The only code that needs to be exported is that of the `lcmath` dependency module.